dnf/base.py

```diff
@@ -139,9 +139,10 @@ def _add_tempfiles(self, files):
 
     def _add_repo_to_sack(self, repo):
         repo.load()
-        mdload_flags = dict(load_filelists=True,
-                            load_presto=repo.deltarpm,
+        mdload_flags = dict(load_presto=repo.deltarpm,
```

Comment:
I am thinking about simplification of the implementation.

The patch introduce storing information about loading filelists in configuration (libdnf MainConf) and demands. Somehow I have a feeling that this structure does not have a benefit and only adds additional layer. Is there any reason why we have to store information in Demands (cannot access main conf but have access to Demands or something else)?

The formal implementation in Hawkey already supported loading filelists as optional, but we did not use that, therefore we don't have to introduce the new path for loading.
The Question is do we have an access to Main Conf in Python Repo Class?

Commands can modify main conf instead of demands
```
        if dnf.util._is_file_pattern_present(self.opts.pkg_specs):
            demands.filelists = True
```

Reply:
That's probably right. I don't see a reason why not to write directly to the configuration. I was mainly keeping the parity here with the existing `repo.load_metadata_other`...